lib/src/config/config.rs

```diff
@@ -397,13 +413,21 @@ impl Config {
     /// # use rocket::config::ConfigError;
     /// # fn config_test() -> Result<(), ConfigError> {
     /// let mut config = Config::new(Environment::Staging)?;
-    /// config.set_port(1024);
+    /// assert!(config.set_port(1024).is_ok());
     /// # Ok(())
     /// # }
     /// ```
     #[inline]
-    pub fn set_port(&mut self, port: u16) {
-        self.port = port;
+    pub fn set_port(&mut self, port: u16) -> Result<()> {
+        match self.address {
+            Address::Tcp(..) => {
+                self.port = Some(port);
+            }
+            Address::Unix(..) => {
+                return Err(ConfigError::BadPort);
```

Comment:
`set_port` now returns an `BadPort` error if address was set to unix socket path.